### ISSUE TITLE
documentation: Forcefully, or manually clear handles with no matching IPs

### ISSUE THREAD

**Issue by zestysoft** [projectcalico/calico#6988]
`calico ipam check` output is stating that I've got handles with no matching IPs that won't go away using this:

```
calicoctl datastore migrate lock
calicoctl ipam check -o report.json
calicoctl ipam release --from-report report.json
calicoctl datastore migrate unlock
```

I'm getting failures for each, like this:
`Deleting handle k8s-pod-network.c4354788e6ab44699b39561a62f3e68baad7f41377e3952fc82000131dfb984d failed: operation Delete is not supported on IPAMHandleKey(id=k8s-pod-network.c4354788e6ab44699b39561a62f3e68baad7f41377e3952fc82000131dfb984d)`

Googling for `calico "no matching ips"` or `"Delete is not supported on IPAMHandleKey"` surprisingly returns 0 results.

**Comment by zestysoft** [comment]
On a whim, I tried `k delete ipamhandle k8s-pod-network.c4354788e6ab44699b39561a62f3e68baad7f41377e3952fc82000131dfb984d` and it deleted.  Hopefully this isn't breaking something.  Is the "Delete is not supported" message above because of some kind of permissions restriction?  I'm curious why it was easily deletable this way.

**Comment by caseydavenport** [comment]
Yeah, I think I see what's up... our calicoctl code is probably calling through to Delete() which isn't supported on IPAM handles.. only DeleteKVP is (which is a special delete method in the backend that handles atomicity better) https://github.com/projectcalico/calico/blob/master/libcalico-go/lib/backend/k8s/resources/ipam_handle.go#L148-L174

This does appear to have been fixed already in this commit: https://github.com/projectcalico/calico/commit/6dced1eb50e602453e068ac8cbcfb4fbf3fb78c5

Which should be available in v3.25.0 and v3.24.2+, as it was cherry-picked here: https://github.com/projectcalico/calico/pull/6652

**Comment by zestysoft** [comment]
Thanks for looking into this!
